### PR TITLE
Add view & edit buttons

### PR DESCRIPTION
**Pull request by royrutto**
Adds view and edit button column for records on table view.

### CODE REVIEW

File: src/hatti/views/table.cljs [hunk 1]
```diff
@@ -82,21 +83,47 @@
   (let [clj-value (js->clj value :keywordize-keys true)]
     (forms/format-answer field clj-value language true)))
 
+(defn button-formatter
+  [row cell value columnDef dataContext]
+  (let [{:keys [owner project formid]} (:dataset-info @shared/app-state)
+        form-owner (last-url-param owner)
+        project-id (last-url-param project)
+        edit-link (str "/"
```
Comment (pld) [inline src/hatti/views/table.cljs]: doesn't the `url` function prefix with "/" already?

File: src/hatti/views/table.cljs [hunk 2]
```diff
@@ -82,21 +83,47 @@
...
+                       (url form-owner project-id formid
+                            "webform?instance-id=")
+                       value)]
+    (str
```
Comment (pld) [inline src/hatti/views/table.cljs]: can we write this in sablono/hiccup syntax and render it to a string?

File: src/hatti/views/table.cljs [hunk 3]
```diff
@@ -115,6 +142,23 @@
...
+  (let [actions (.getElementsByClassName js/document "record-actions")
+        sg-viewport (first (.getElementsByClassName js/document
+                                                    "slick-viewport"))]
+
+    (doseq [action actions]
+      (let [leftOffset (js/parseInt (.-offsetLeft action))]
+        (.addEventListener sg-viewport "scroll"
+                           (fn []
+                             (set! (.-left (.-style action))
+
```
Comment (pld) [inline src/hatti/views/table.cljs]: remove newline

File: test/hatti/views/table_test.cljs [hunk 4]
```diff
@@ -95,7 +96,17 @@
...
         ths #(sel (sel1 % :.slick-header-columns) :.slick-header-column)
         htexts (fn [table] (->> table ths (map dommy/text)))
-        htitles (fn [table] (->> table ths (map #(dommy/attr % :title))))]
+        htitles (fn [table] (->> table ths (map #(dommy/attr % :title))))
+        edit-urls (fn [d]
+                    (let [{:keys [owner project formid]}
+                          (:dataset-info @shared/app-state)
+                          form-owner (last-url-param owner)
+                          project-id (last-url-param project)
+                          edit-link (str "/"
```
Comment (pld) [inline test/hatti/views/table_test.cljs]: "/" again

File: src/hatti/views/table.cljs (hunk 1 above)
Comment (royrutto) [inline src/hatti/views/table.cljs]: the `url` defined in hatti doesn't https://github.com/onaio/hatti/blob/master/src/hatti/utils.cljs#L7 , adding it breaks other stuff that uses it.

Will modify what uses it to factor in the already prefixed "/" , similar to the one we use in zebra https://github.com/onaio/milia/blob/master/src/milia/utils/url.cljc#L13

Comment (pld) [inline src/hatti/views/table.cljs]: oh boo, might be time to pull the utils out of hatti and milia (and any extra stuff in zebra) and make a clj-utils library that's used and uniform across hatti, milia, and zebra

File: src/hatti/views/table.cljs (hunk 2 above)
Comment (royrutto) [inline src/hatti/views/table.cljs]: yes, @okal wrote a function for this in Zebra but we'll need to copy it to hatti utils to make use if it https://github.com/onaio/zebra/blob/master/src/ona/utils/walkthrough.cljc#L55

I agree, I think we should create the `clj-utils` lib for such util functions.

File: project.clj [hunk 5]
```diff
@@ -3,7 +3,7 @@
       [path]
       (str "resources/public/js/lib/" path))
 
-(defproject onaio/hatti "0.1.21"
+(defproject onaio/hatti "0.1.21-SNAPSHOT"
```
Comment (pld) [inline project.clj]: if 1.21 is the release, we should put this in 1.22-snapshot

**Pull request merged**